GitCommands/CommitDataManager.cs

```
@@ -135,7 +135,7 @@ internal CommitData CreateFromFormattedData(string data)
             var guid = ObjectId.Parse(lines[0]);
 
             // TODO: we can use this to add more relationship info like gitk does if wanted
-            var treeGuid = ObjectId.Parse(lines[1]);
+            ObjectId? treeGuid = string.IsNullOrWhiteSpace(lines[1]) ? null : ObjectId.Parse(lines[1]);
```

TryParse()?
```suggestion
            ObjectId.TryParse(lines[1], out ObjectId? treeGuid);
```

---

`TryParse` was my first thought (refer to the branch name).
Though it would hide errors.
The bug report was about an empty string.

---

Personally, I would prefer the simpler code and ignore malformed ids (like error messages accidentally here)

---

> Personally, I would prefer the simpler code and ignore malformed ids (like error messages accidentally here)

`TryParse` would hide errors.
It *is* worth to make all expectations explicit.
If an exception from this rule is made, I want to have a good, written reason. I have seen "This error cannot happen" nevertheless occurring too often.
Either this line is empty or it contains a valid id.
Rather I would remove this unused property than `TryParse` it.

---

Or else add a method `ParseOptional` - perhaps with a nicer name.

---

Shall I remove the unused property `CommitData.TreeGuid`?
Or else, shall I add a method `ObjectId.ParseIfNotEmpty`?

---

> Shall I remove the unused property `CommitData.TreeGuid`?

Yes, if needed it can be added again.
CommitInfo currently have no need of the commit contents

---

Removed the property `CommitData.TreeGuid` and updated the PR title and description.